Trellix/trellix-epo-on-prem/ingest/parser.yml

```
@@ -0,0 +1,116 @@
+
+          host.mac: "{{parsed_event.output.EPOevent.MachineInfo.RawMACAddress}}"
+          host.ip: "{{parsed_event.output.EPOevent.MachineInfo.IPAddress}}"
+          host.os.full: "{{parsed_event.output.EPOevent.MachineInfo.OSName}}"
+
+          user.name: "{{parsed_event.output.EPOevent.MachineInfo.UserName}}"
+
+          event.severity: "{{parsed_event.output.EPOevent.SoftwareInfo.Event.Severity}}"
+          event.provider: "{{parsed_event.output.EPOevent.SoftwareInfo.Event.ProductName}}"
+          event.action: "{{parsed_event.output.EPOevent.SoftwareInfo.Event.CommonFields.ThreatActionTaken}}"
+
+          file.name: "{{parsed_event.output.EPOevent.SoftwareInfo.Event.CommonFields.TargetFileName}}"
+          process.name: "{{parsed_event.output.EPOevent.SoftwareInfo.Event.CommonFields.TargetProcessName}}"
+          user.target.name: "{{parsed_event.output.EPOevent.SoftwareInfo.Event.CommonFields.TargetUserName}}"
+
+          threat.indicator.name: "{{parsed_event.output.EPOevent.SoftwareInfo.Event.CommonFields.ThreatName}}"
+          threat.indicator.sightings: "{{parsed_event.output.EPOevent.SoftwareInfo.Event.CommonFields.ThreatHandled}}"
+
+          trellix.event.id: "{{parsed_event.output.EPOevent.SoftwareInfo.Event.EventID}}"
+          trellix.event.detect_date: "{{parsed_event.output.EPOevent.SoftwareInfo.Event.CommonFields.DetectedUTC | to_rfc3339}}"
+
+          trellix.threat.action_taken: "{{parsed_event.output.EPOevent.SoftwareInfo.Event.CommonFields.ThreatActionTaken}}"
+          trellix.threat.severity: "{{parsed_event.output.EPOevent.SoftwareInfo.Event.CommonFields.ThreatSeverity}}"
+          trellix.threat.category: "{{parsed_event.output.EPOevent.SoftwareInfo.Event.CommonFields.ThreatCategory}}"
+          trellix.threat.type: "{{parsed_event.output.EPOevent.SoftwareInfo.Event.CommonFields.ThreatType}}"
+        filter: "{{parsed_event.output.get('EPOevent') != None}}"
+
+  set_update_events_fields:
+    actions:
+      - set:
+          "@timestamp": "{{parsed_event.output.UpdateEvents.McAfeeCommonUpdater.UpdateEvent.GMTTime}}"
+
+          agent.id: "{{parsed_event.output.UpdateEvents.MachineInfo.AgentGUID.strip('{}')}}"
+          agent.version: "{{parsed_event.output.UpdateEvents.MachineInfo.AgentVersion}}"
+
+          host.mac: "{{parsed_event.output.UpdateEvents.MachineInfo.RawMACAddress}}"
+          host.ip: "{{parsed_event.output.UpdateEvents.MachineInfo.IPAddress}}"
+          host.os.full: "{{parsed_event.output.UpdateEvents.MachineInfo.OSName}}"
+
+          event.code: "{{parsed_event.output.UpdateEvents.McAfeeCommonUpdater.UpdateEvent.EventID}}"
+          event.reason: "{{parsed_event.output.UpdateEvents.McAfeeCommonUpdater.UpdateEvent.Description}}"
+          event.severity: "{{parsed_event.output.UpdateEvents.McAfeeCommonUpdater.UpdateEvent.Severity}}"
+
+          trellix.update_event.user_name: "{{parsed_event.output.UpdateEvents.MachineInfo.UserName}}"
```

As this field can contain several user_name, I suggest renaming it and split its content:
```suggestion
          trellix.update_event.usernames: "{{parsed_event.output.UpdateEvents.MachineInfo.UserName.split(','}}"
```